server/core/src/https/middleware/mod.rs

```diff
@@ -73,3 +82,93 @@ pub async fn kopid_middleware(mut request: Request<Body>, next: Next) -> Respons
 
     response
 }
+
+// This middleware extracts the ip_address and client information, and stores it
+// in the request extensions for future layers to use it.
+pub async fn ip_address_middleware(
+    State(state): State<ServerState>,
+    mut request: Request<Body>,
+    next: Next,
+) -> Response {
+    match ip_address_middleware_inner(&state, &mut request).await {
+        Ok(trusted_client_ip) => {
+            // By this point, proxy-v2 AND x-forward-for have resolved, so we can finally display this information.
+            info!(connection_addr = %trusted_client_ip.connection_addr, client_ip_addr = %trusted_client_ip.client_ip_addr);
+            request.extensions_mut().insert(trusted_client_ip);
+            next.run(request).await
+        }
+        Err((status_code, reason)) => {
+            // Worst case, return.
+            let mut response = Response::new(Body::from(reason));
+            *response.status_mut() = status_code;
+            response
+        }
```

Comment:
This can just use the `axum::response::IntoResponse` trait if you import the trait.
```suggestion
        Err(err) => err.into_response()
```